lib/jbuilder.rb

```diff
@@ -276,6 +238,55 @@ def target!
 
   alias_method :method_missing, :set!
 
+  def _set(key, value = BLANK, attributes = nil, &block)
+    result = if block
+      if _blank?(value)
+        # json.comments { ... }
+        # { "comments": ... }
+        _merge_block key, &block
+      else
+        # json.comments @post.comments { |comment| ... }
+        # { "comments": [ { ... }, { ... } ] }
+        _scope { _array value, &block }
+      end
```

Comment:
This condition used to be the inverse, with a `if !_blank?`. I think this improves control flow, and it saves a tiny bit in processing.